sql/src/main/java/io/crate/execution/dml/ShardResponse.java

```diff
@@ -139,11 +182,37 @@ public ShardResponse(StreamInput in) throws IOException {
         if (in.readBoolean()) {
             failure = in.readException();
         }
+        if (allOn4_2) {
+            int resultColumnsSize = in.readVInt();
+            if (resultColumnsSize > 0) {
+                resultColumns = new Symbol[resultColumnsSize];
+                for (int i = 0; i < resultColumnsSize; i++) {
+                    Symbol symbol = Symbols.fromStream(in);
+                    resultColumns[i] = symbol;
+                }
+                Streamer[] resultRowStreamers = Symbols.streamerArray(List.of(resultColumns));
+                int resultRowsSize = in.readVInt();
+                if (resultRowsSize > 0) {
+                    resultRows = new ArrayList<>(resultRowsSize);
+                    int rowLength = in.readVInt();
+                    for (int i = 0; i < resultRowsSize; i++) {
+                        Object[] row = new Object[rowLength];
+                        for (int j = 0; j < rowLength; j++) {
+                            row[j] = resultRowStreamers[j].readValueFrom(in);
+                        }
+                        resultRows.add(row);
+                    }
+                }
+            }
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
+        if (out.getVersion().onOrAfter(Version.V_4_2_0)) {
+            out.writeBoolean(allOn4_2);
```

Comment:
the version is already inside the stream output, why do we have to stream the version hint again using a boolean?

Reply:
You are right, i will open a new pr for that.